src/core/Akka.Remote.Tests/RemotingSpec.cs

```diff
@@ -457,6 +475,171 @@ public async Task Bug_884_Remoting_must_support_reply_to_child_of_Routee()
+                    (new ActorAssociationEventListener(remoteTransportProbe)));
+
+                // Hijack associations through the test transport
+                AwaitCondition(() => registry.TransportsReady(rawLocalAddress, rawRemoteAddress));
+                var testTransport = registry.TransportFor(rawLocalAddress).Item1;
+                testTransport.WriteBehavior.PushConstant(true);
+
+                // Force an outbound associate on the real system (which we will hijack)
+                // we send no handshake packet, so this remains a pending connection
+                var dummySelection = thisSystem.ActorSelection(ActorPath.Parse(remoteAddress + "/user/noonethere"));
+                dummySelection.Tell("ping", Sys.DeadLetters);
+
+                var remoteHandle = remoteTransportProbe.ExpectMsg<InboundAssociation>(TimeSpan.FromMinutes(4));
+                remoteHandle.Association.ReadHandlerSource.TrySetResult((IHandleEventListener)(new ActionHandleEventListener(ev => { })));
+
+                // Now we initiate an emulated inbound connection to the real system
+                var inboundHandleProbe = CreateTestProbe();
+                var inboundHandleTask = remoteTransport.Associate(rawLocalAddress);
+                inboundHandleTask.Wait(TimeSpan.FromSeconds(3));
+                var inboundHandle = inboundHandleTask.Result;
+                inboundHandle.ReadHandlerSource.SetResult(new ActorHandleEventListener(inboundHandleProbe));
+
+                AwaitAssert(() =>
+                {
+                    registry.GetRemoteReadHandlerFor(inboundHandle.AsInstanceOf<TestAssociationHandle>()).Should().NotBeNull();
+                });
+
+                var pduCodec = new AkkaPduProtobuffCodec(Sys);
+
+                var handshakePacket = pduCodec.ConstructAssociate(new HandshakeInfo(remoteAddress, 0));
+                var brokenPacket = pduCodec.ConstructPayload(ByteString.CopyFrom(0, 1, 2, 3, 4, 5, 6));
+
+                // Finish the inbound handshake so now it is handed up to Remoting
+                inboundHandle.Write(handshakePacket);
+                // Now bork the connection with a malformed packet that can only signal an error if the Endpoint is already registered
+                // but not while it is stashed
+                inboundHandle.Write(brokenPacket);
+
+                // No disassociation now - the connection is still stashed
+                inboundHandleProbe.ExpectNoMsg(1000);
+
+                // Finish the handshake for the outbound connection - this will unstash the inbound pending connection.
+                remoteHandle.Association.Write(handshakePacket);
+
+                inboundHandleProbe.ExpectMsg<Disassociated>(TimeSpan.FromMinutes(5));
+            }
+            finally
+            {
+                Shutdown(thisSystem);
+            }
+        }
+
+        [Fact]
+        public void Properly_quarantine_stashed_inbound_connections()
```

Comment:
Additional spec brought from JVM - covers other half of the reproducers for this issue.